packages/api-plugin-fulfillment/src/mutations/createFulfillmentMethod.js

```diff
@@ -0,0 +1,59 @@
+import SimpleSchema from "simpl-schema";
+import Random from "@reactioncommerce/random";
+import ReactionError from "@reactioncommerce/reaction-error";
+import { FulfillmentMethodSchema } from "../simpleSchemas.js";
+
+const inputSchema = new SimpleSchema({
+  method: FulfillmentMethodSchema,
+  fulfillmentTypeId: String,
+  shopId: String
+});
+
+/**
+ * @method createFulfillmentMethodMutation
+ * @summary Creates a fulfillment method
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - Input (see SimpleSchema)
+ * @returns {Promise<Object>} An object with a `method` property containing the created method
+ */
+export default async function createFulfillmentMethodMutation(context, input) {
+  const cleanedInput = inputSchema.clean(input); // add default values and such
+  inputSchema.validate(cleanedInput);
+
+  const { method: inputMethod, fulfillmentTypeId, shopId } = cleanedInput;
+  const { collections } = context;
+  const { Fulfillment } = collections;
+  const method = { ...inputMethod };
+
+  await context.validatePermissions("reaction:legacy:fulfillmentMethods", "create", { shopId });
+
+  const ffTypeRecord = await Fulfillment.findOne({ _id: fulfillmentTypeId, shopId });
+  if (!ffTypeRecord) throw new ReactionError("server-error", "Unable to create fulfillment method without defined type");
+
+  const ffTypeMethodRecord = await Fulfillment.findOne({
```

Comment:
The _id only gets us the ff-type. Along with the ff-type, I am querying for the specific ff-method also before adding a new ff-method.

Reply:
This answer doesn't actually help clarify anything. Whatever you query with that _id is always going to give you the exact same record.

Reply:
Yes, we get the exact same ff-type record. But if the other parameters related to ff-method also match, then we need not proceed further since the new record is going to be a duplicate ff-method and we return error.
Else we should be further checking within the methods array of the returned record to see if we already have the new method

Reply:
But I think we should remove the matching with name as you mentioned in a different review comment and retain only fulfillmentMethod matching. Else it would allow same method with different name

`methods: { $elemMatch: { fulfillmentMethod: method.fulfillmentMethod } }`

Reply:
@zenweasel Please let me know if this can be considered resolved